Source/DafnyStandardLibraries/examples/DynamicArrayExamples.dfy

```diff
@@ -0,0 +1,52 @@
+module DynamicArrayExamples {
+  import opened DafnyStdLibs.DynamicArray
+  import opened DafnyStdLibs.BoundedInts
+
+  method {:test} InitialCapacity() {
+    var arr := new DynamicArray<int>(0, 101);
+    for i: int := 0 to 100
+      invariant fresh(arr.Repr)
+      invariant arr.Valid?()
+      invariant arr.size as int == i
+      invariant arr.capacity == 101
+    {
+      arr.PushFast(i);
+    }
+  }
+
+  method {:test} Ensure() {
+    var arr := new DynamicArray<int>(0);
+    assert arr.size == 0;
+    arr.Ensure(101);
+    assert arr.capacity >= 101;
+    assert arr.size == 0;
+    for i: int := 0 to 100
```

Comment:
Same comments for this test. The `: int` in the `for` loop is unnecessary, and you could (if you want to) change the `101` to `100`.